src/charm.py

```diff
@@ -231,6 +236,13 @@ def _on_collect_unit_status(self, event: ops.CollectStatusEvent) -> None:
 
         event.add_status(ActiveStatus())
 
+    def snap_channel_changed(self) -> bool:
```

Comment:
**nitpick (blocking):** 

The snap version information shouldn't be stored in the local stored. It's the information you can get in live time.